relayer/provider/provider.go

```diff
@@ -46,6 +47,43 @@ type ChainProvider interface {
 	ImportKeystore(context.Context, string, string) (string, error)
 	RevertMessage(context.Context, *big.Int) error
 	GetFee(context.Context, string, bool) (uint64, error)
-	SetFee(context.Context, string, uint64, uint64) error
+	SetFee(context.Context, string, *big.Int, *big.Int) error
 	ClaimFee(context.Context) error
 }
+
+// CommonConfig is the common configuration for all chain providers
+type CommonConfig struct {
```

Comment:
Creating CommonConfig will do us more harm than benefit. It results in more rigid and coupled system among the chains thereby making us difficult to make changes in the future.

Reply:
Common config can be way of reusing common functionality of all the chains, basically OOP.